docs/developer-docs/latest/development/custom-fields.md

```diff
@@ -0,0 +1,340 @@
+---
+title: Custom fields reference - Strapi Developer Docs
+description: Learn how you can use custom fields to extend Strapi's content-types capabilities.
+sidebarDepth: 3
+canonicalUrl: https://docs.strapi.io/developer-docs/latest/development/custom-fields.html
+---
+
+# Custom fields
+
+<!-- TODO: add links to user guide section(s) somewhere -->
+Custom fields extend Strapi’s capabilities by adding new types of fields to content-types. Once created or installed, custom fields can be used in the Content-Types Builder and Content Manager just like built-in fields.
+
+The present documentation is intended to custom fields creators: it describes which APIs and functions developers must use to create a new custom field. The [user guide](#) describes how to install and use custom fields from Strapi's admin panel.
+
+<!-- TODO: uncomment and adjust content when blog post is ready -->
+<!-- ::: strapi Prefer to learn by building?
+If you'd rather directly jump to a concrete example, see our [Creating a color custom field guide](#) page for step-by-step instructions on how to build your first custom field from scratch.
+::: -->
+
+The recommended way to create custom fields is through creating [plugins](/developer-docs/latest/development/plugins-development.md) that include both a server and an admin panel part. Both parts should be registered separately before a custom field is usable in Strapi's admin panel.
+
+Once created and used, in the attributes defined in `schema.json` files for the models, custom fields are defined as just another type of field (i.e. `type: 'custom-field'`) with a few additional attributes depending on the custom field used (see [models documentation](/developer-docs/latest/development/backend-customization/models.md#custom-fields)).
+<!-- The section linked above will be part of another PR, but basically it's about mentioning that in schema files, custom fields are defined with `type: 'custom-field'` and a possible additional `customField` key for the uid (`global::…` or `plugin::…`), plus possibly additional keys (e.g. `options` for a `select` setting, etc.) -->
+
+::: note
+Though the recommended way to add a custom field is through creating a plugin, app-specific custom fields can also be registered within the global `bootstrap` and `register` [functions](/developer-docs/latest/setup-deployment-guides/configurations/optional/functions.md) found in `./src/index.js`.
+:::
+
+<!-- TODO: make sure this # exists in the backend custom. > models docs — will come in another PR -->
+
+<!-- ? should we document these? 👇 this was described in the [technical RFC](https://github.com/strapi/rfcs/blob/3eb034e169746558315d719ca5fb49cec854640a/rfcs/xxxx-custom-fields-api.md#motivation) but I'm unsure about what to do with it -->
```

Comment:
I don't think they're necessary in the docs, the intent was only to context on the RFC to get higher quality feedback

Reply:
Thank you, Rémi!

Based on the other comments, it seems like it's important to mention that some data types aren't supported, and that custom fields can only be shared with plugins.

So I think it would be worth keeping this callout here to set readers' expectations right from the start. What do you think?

Reply:
Sounds good to me :)